py/picca/fitter2/effective-bins.py

```diff
@@ -0,0 +1,186 @@
+        pars[i] = f['best fit'].attrs[i][0]
+        err_pars[i] = 0
+    xi= sp.array(f['LYA(LYA)-LYA(LYA)']['fit'])
+    return free_p,fixed_p,pars,err_pars,xi
+
+def extract_data(chi2file,dic_init):
+    cp = ConfigParser.ConfigParser()
+    cp.optionxform=str
+    cp.read(chi2file)
+    dic_data = fit_parser.parse_data(cp.get('data sets','ini files'),cp.get('data sets','zeff'),dic_init['fiducial'])
+    data_file = dic_data['data']['filename']
+    f = fitsio.FITS(data_file)
+    cov = f[1]["CO"][:]
+    rp  = f[1]["RP"][:]
+    rt  = f[1]["RT"][:]
+    z   = f[1]["Z"][:]
+    ico = linalg.inv(cov)
+
+    head = f[1].read_header()
+    nt = head['NT']
+    np = head['NP']
+    rt_min = 0
+    rt_max = head['RTMAX']
+    rp_min = head['RPMIN']
+    rp_max = head['RPMAX']
+
+    r_min = dic_data['cuts']['r-min']
+    r_max = dic_data['cuts']['r-max']
+    return ico,rp,rt,np,nt,rp_min,rp_max,rt_min,rt_max,r_min,r_max,z
+
+def apply_mask(dm,ico,z,dic):
+    for d in dic['data sets']['data']:
+        for i in dm:
+            dm_dp[i][d.mask==0] = 0
+        ico[:,d.mask==0] = 0
+        ico[d.mask==0,:] = 0
+        z[d.mask==0] = 0
+    
+def xi_mod(pars,dic_init):
+    k = dic_init['fiducial']['k']
+    pk_lin = dic_init['fiducial']['pk']
+    pksb_lin = dic_init['fiducial']['pksb']
+    for d in dic_init['data sets']['data']:
+        pars['SB'] = False
+        xi_best_fit = pars['bao_amp']*d.xi_model(k, pk_lin-pksb_lin, pars)
+        pars['SB'] = True
+        snl = pars['sigmaNL_per']
+        pars['sigmaNL_per'] = 0
+        xi_best_fit += d.xi_model(k, pksb_lin, pars)
+        pars['sigmaNL_per'] = snl
+        pars['SB'] = False
+        
+    return xi_best_fit
+    
+def plot_xi(xi,title = ' '):
+    pylab.figure()
+    pylab.imshow(xi.reshape((np,nt)),origin=0,interpolation='nearest',extent=[rt_min,rt_max,rp_min,rp_max])
+    pylab.colorbar()
+    pylab.ylabel(r"$r_{\parallel}$",size=20)
+    pylab.xlabel(r"$r_{\perp}$",size=20)
+    pylab.title(title,size = 20)
+
+def xi_mod_p(x,pname,pars):
+    pars2 = copy.deepcopy(pars)
+    pars2[pname] = x
+    return xi_mod(pars2,dic_init)
+
+def compute_dm_dp(freep,pars):
+    dm = {}
+    for p in freep :
+        g = functools.partial(xi_mod_p,pname=p,pars=pars)
+        dm[p] = derivative(g,pars[p])
+    return dm
+
+def compute_M(dm_dp,icov):
+    M = sp.zeros(icov.shape)
+    for i in range(icov.shape[0]):
+        for j in range(icov.shape[0]):
+            M[i,j] = dm_dp[i] * icov[i,j] * dm_dp[j]
+    return M
+
+def compute_z0(M,z):
+    res = 0
+    den = 0
+    for i in range(z.shape[0]):
+        for j in range(z.shape[0]):
+            res += M[i,j]*z[i]
+            den += M[i,j]
+    return res,den
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument('--chi2-file', type = str, default = None, required=True,
+                        help = 'chi2 input file name')
```

Comment:
Correct spacing and describe better. Something like "Path to the config 'chi2.ini' file used in fitter2"